### PR TITLE
Update to gridicons 3

### PR DESCRIPTION
**Pull request by oandregal**
This is the companion PR for https://github.com/Automattic/gridicons/pull/283

It updates the Gridicons dependency to v3 line, which includes the ability to import individual icons and changes the package paths (from `build/` to `dist/`).

Note that this PR doesn't change how we import Gridicons except for the Gridicons example.

### Testing

**General testing**

* any section that contains gridicons and verify that they work as intended.
* http://calypso.localhost:3000/devdocs/design and verify that gridicons are published and clickable.

**Change some icon to use the individual import and verify that it works**

For example, go to `client/layout/masterbar/item.js` and:

    // change the Gridicon import to:
    import GridiconReader from 'gridicons/dist/reader';
    ...
    // add a new function within the Masterbar component to test that onClick works
    onClick() { 
	window.alert( 'clicked the Reader Gridicon' );
    }
    ...
    // substitute the old Gridicon for:
    { !! this.props.icon && (
        <GridiconReader
	    size={ 18 }
	    className={ 'test-class' }
	    onClick={ this.onClick }
	    myprop={ 'my prop' }
	    />
        ) }

The expected result is that on visiting http://calypso.localhost:3000/ the SVG element:

* has the classes: `gridicon`, `gridicons-reader`, `test-class`, `needs-offset-x`
* its height and width are 18
* contains the `myprop` property
* on click shows an alert

### CODE REVIEW

File: client/devdocs/design/index.jsx
Comment (ockham) [inline client/devdocs/design/index.jsx]: Since we're doing it already for `social-logos`, let's add an alias for Gridicon to our webpack config (as seen [here](https://github.com/Automattic/wp-calypso/pull/23086/files#diff-11e9f7f953edc64ba14b0cc350ae7b9dR171)).

Comment (ockham) [reply client/devdocs/design/index.jsx]: (That will allow us to `import Gridicons from 'gridicons/example';` here.)

Comment (nosolosw) [reply client/devdocs/design/index.jsx]: Done at 166ed8a

File: package.json
Comment (ockham) [inline package.json]: Ideally, the prereleases should’ve been `3.0.0-alpha.1`, `3.0.0-alpha.2`. I think we can and should still have `3.0.0` be the final release.

Comment (ockham) [reply package.json]: Looks like we can `npm unpublish gridicons@3.0.0-alpha.1` and `3.0.0-alpha.2`

Comment (nosolosw) [reply package.json]: For the record, the mess I provoked publishing wrong alphas has been solved and [documented](https://github.com/Automattic/gridicons/pull/283/commits/31ccaf0c436496df7d13e3515ab54ed25bdb7988):

* unpublished gridicons 3.0.1-alpha and 3.0.2-alpha from npm (so they can be reused in the future)
* published gridicons 3.0.0-alpha.1 to npm
* changed the dependency here to 3.0.0-alpha.1

Comment (gwwar) [reply package.json]: Careful with `npm unpublish` we generally only use this when we've accidentally published something that we didn't intend to. The related version cannot be reused in the future.

File: npm-shrinkwrap.json
Comment (gwwar) [inline npm-shrinkwrap.json]: Note that fsevents is an optional dependency. So I suppose we'll see thrashing, whenever someone using OSX vs anything else updates the shrinkwrap. https://github.com/npm/npm/issues/17722

**Comment by gwwar** [comment]
Looking pretty good! Should we update the example text to prefer importing the component directly?

screen shot 2018-03-09 at 4 56 19 pm

@Automattic/flowpatrol would it be possible to get some help spot checking for gridicons 🐛 on this branch? ❤️

**Comment by brbrr** [comment]
Branch need rebase to make tests pass

**Comment by brbrr** [comment]
automated and manual tests did not find any issues 👍

**Comment by nosolosw** [comment]
@gwwar re: https://github.com/Automattic/wp-calypso/pull/23171#issuecomment-371987807

Perhaps that's worth doing when Calypso itself is using the new icons? At the moment Calypso ships the whole iconset, so until we change that to use individual icons it won't provide any size reduction (it actually could make it bigger because Calypso will ship both, the whole iconset _and_ the individual icon).

**Comment by gwwar** [comment]
> Perhaps that's worth doing when Calypso itself is using the new icons?

Sure that sounds fine to me, just as long as we remember to follow up :)

> it actually could make it bigger because Calypso will ship both

We can verify any bundle size changes by using @jsnajdr ✨tool:
http://iscalypsofastyet.com/branch?branch=update/gridicons-3

This branch has negligible changes.

```
Commit: c70662faad816fb7c41397469ae6a6934e584939
Author: nosolosw
At: 2018-03-12T06:43:45Z
Message: Update shrinkwrap
Delta:
chunk                                                       stat_size           parsed_size           gzip_size
accept-invite                                                    +0 B                  -2 B  (-0.0%)       -2 B  (-0.0%)
ads                                                              +0 B                  -8 B  (-0.0%)       -2 B  (-0.0%)
async-load-blocks-calendar-popover                               +0 B                  -2 B  (-0.0%)       -2 B  (-0.0%)
async-load-blocks-reader-full-post                               +0 B                  -4 B  (-0.0%)       -2 B  (-0.0%)
async-load-components-happychat                                  +0 B                  -2 B  (-0.0%)       -2 B  (-0.0%)
async-load-components-web-preview-component                      +0 B                 -28 B  (-0.0%)       +3 B  (+0.0%)
async-load-design                                              -137 B  (-0.0%)       -158 B  (-0.0%)      -20 B  (-0.0%)
async-load-design-blocks                                         +0 B                 -70 B  (-0.0%)       -3 B  (-0.0%)
async-load-extensions-woocommerce-app-store-stats                +0 B                 -10 B  (-0.0%)       -2 B  (-0.0%)
async-load-extensions-woocommerce-app-store-stats-listview       +0 B                  -4 B  (-0.0%)       -2 B  (-0.0%)
async-load-featured-image                                        +0 B                  -2 B  (-0.1%)       -2 B  (-0.2%)
async-load-layout-masterbar-drafts-popover                       +0 B                  -2 B  (-0.0%)       -3 B  (-0.1%)
async-load-my-sites-guided-transfer                              +0 B                  -2 B  (-0.0%)       -1 B  (-0.0%)
async-load-my-sites-site-settings-section-export                 +0 B                  -4 B  (-0.0%)       -2 B  (-0.0%)
async-load-my-sites-site-settings-section-import                 +0 B                  -8 B  (-0.0%)       +2 B  (+0.0%)
async-load-post-editor-editor-author                             +0 B                  -2 B  (-0.0%)       -2 B  (-0.0%)
async-load-post-editor-editor-location                           +0 B                  -2 B  (-0.0%)       -2 B  (-0.1%)
async-load-post-editor-editor-post-formats-accordion             +0 B                  -2 B  (-0.0%)       -2 B  (-0.1%)
async-load-post-editor-editor-sharing-accordion                  +0 B                  -2 B  (-0.0%)       -2 B  (-0.0%)
async-load-post-editor-media-modal                               +0 B                 -20 B  (-0.0%)       -3 B  (-0.0%)
async-load-reader-conversations-stream                           +0 B                  -2 B  (-0.0%)       -4 B  (-0.2%)
async-load-reader-feed-stream                                    +0 B                 -10 B  (-0.0%)       -2 B  (-0.0%)
async-load-reader-following-manage                               +0 B                 -12 B  (-0.0%)       -2 B  (-0.0%)
async-load-reader-list-stream                                    +0 B                  -2 B  (-0.0%)       -2 B  (-0.1%)
async-load-reader-search-stream                                  +0 B                  -6 B  (-0.0%)       -2 B  (-0.0%)
async-load-reader-sidebar                                        +0 B                  -2 B  (-0.0%)       -2 B  (-0.0%)
async-load-reader-site-stream                                    +0 B                 -10 B  (-0.0%)       -2 B  (-0.0%)
async-load-reader-tag-stream-main                                +0 B                  -4 B  (-0.0%)       -2 B  (-0.0%)
build                                                            +0 B                 -40 B  (-0.0%)       +3 B  (+0.0%)
checklist                                                        +0 B                  -4 B  (-0.0%)       +0 B
checkout                                                         +0 B                 -18 B  (-0.0%)       -9 B  (-0.0%)
comments                                                         +0 B                 -14 B  (-0.0%)       -2 B  (-0.0%)
concierge                                                        +0 B                  -6 B  (-0.0%)       -2 B  (-0.0%)
customize                                                        +0 B                  -2 B  (-0.0%)       -1 B  (-0.0%)
devdocs                                                          +0 B                 -22 B  (-0.0%)       +9 B  (+0.0%)
domain-connect-authorize                                         +0 B                  -2 B  (-0.0%)       -1 B  (-0.0%)
domains                                                          +0 B                 -28 B  (-0.0%)      -16 B  (-0.0%)
google-my-business                                               +0 B                  -4 B  (-0.0%)       -4 B  (-0.1%)
happychat                                                        +0 B                 -14 B  (-0.0%)       -2 B  (-0.0%)
help                                                             +0 B                 -24 B  (-0.0%)       -4 B  (-0.0%)
jetpack-connect                                                  +0 B                 -18 B  (-0.0%)       -5 B  (-0.0%)
jetpack-onboarding                                               +0 B                  -4 B  (-0.0%)       -1 B  (-0.0%)
login                                                            +0 B                  -6 B  (-0.0%)       +5 B  (+0.0%)
mailing-lists                                                    +0 B                  -2 B  (-0.0%)       -2 B  (-0.1%)
manifest                                                         +0 B                  +0 B                -1 B  (-0.0%)
me                                                               +0 B                 -14 B  (-0.0%)       -2 B  (-0.0%)
media                                                            +0 B                 -22 B  (-0.0%)       -3 B  (-0.0%)
notification-settings                                            +0 B                 -20 B  (-0.0%)       -5 B  (-0.0%)
people                                                           +0 B                 -18 B  (-0.0%)       -2 B  (-0.0%)
plans                                                            +0 B                 -22 B  (-0.0%)       -8 B  (-0.0%)
plugins                                                          +0 B                 -34 B  (-0.0%)       -2 B  (-0.0%)
post-editor                                                      +0 B                 -56 B  (-0.0%)       -3 B  (-0.0%)
posts-custom                                                     +0 B                 -26 B  (-0.0%)       +7 B  (+0.0%)
posts-pages                                                      +0 B                 -30 B  (-0.0%)       +8 B  (+0.0%)
preview                                                          +0 B                  -2 B  (-0.0%)       +0 B
privacy                                                          +0 B                 -14 B  (-0.0%)       -3 B  (-0.0%)
purchases                                                        +0 B                 -28 B  (-0.0%)       -6 B  (-0.0%)
reader                                                           +0 B                 -32 B  (-0.0%)       -2 B  (-0.0%)
security                                                         +0 B                 -26 B  (-0.0%)       -6 B  (-0.0%)
settings                                                         +0 B                 -22 B  (-0.0%)       -5 B  (-0.0%)
settings-discussion                                              +0 B                  -8 B  (-0.0%)       -2 B  (-0.0%)
settings-security                                                +0 B                 -14 B  (-0.0%)       -3 B  (-0.0%)
settings-traffic                                                 +0 B                 -12 B  (-0.0%)       -5 B  (-0.0%)
settings-writing                                                 +0 B                 -22 B  (-0.0%)       -2 B  (-0.0%)
sharing                                                          +0 B                 -14 B  (-0.0%)       +0 B
signup                                                           +0 B                 -28 B  (-0.0%)      -10 B  (-0.0%)
stats                                                            +0 B                 -28 B  (-0.0%)       +4 B  (+0.0%)
theme                                                            +0 B                 -16 B  (-0.0%)       -2 B  (-0.0%)
themes                                                           +0 B                 -20 B  (-0.0%)       -2 B  (-0.0%)
vendor                                                        -2333 B  (-0.1%)      -2019 B  (-0.2%)     -720 B  (-0.3%)
woocommerce                                                      +0 B                 -32 B  (-0.0%)       -3 B  (-0.0%)
wp-job-manager                                                   +0 B                 -10 B  (-0.0%)       +0 B
wp-super-cache                                                   +0 B                 -10 B  (-0.0%)       -1 B  (-0.0%)
zoninator                                                        +0 B                  -8 B  (-0.0%)       -2 B  (-0.0%)
account                                                          +0 B                 -20 B  (-0.0%)       -4 B  (-0.0%)
```

**Comment by nosolosw** [comment]
> This branch has negligible changes.

Yup, because it doesn't change how do we use Gridicons, it just updates to the latest version to verify that it still works as intended using the general import.

Update Calypso to use the individual icons may be another mini-hack project, and I don't think I'll have the time to do it in the coming weeks. I'd prefer others to take the lead here - and reap the benefits from the size reduction that future PR will provide! :)

**Comment by gwwar** [comment]
> I'd prefer others to take the lead here

This was why I pinged @Automattic/team-calypso as a heads up :) there's likely a good chance for bundle size reduction here. Your approach here has been great so far @nosolosw ! Incremental changes are easier to review and 🚢

File: client/jetpack-connect/test/__snapshots__/authorize.js.snap
Comment (samouri) [inline client/jetpack-connect/test/__snapshots__/authorize.js.snap]: why is this file being changed in this PR?

**Comment by blowery** [comment]
Thanks for doing this @nosolosw! It's been on my radar for a while now. We can certainly take the lead on moving everything else over.

**Comment by nosolosw** [comment]
This has been updated to the latest gridicons alpha, rebased and squashed.

**Review by ockham — APPROVED** [review]
> The expected result is that on visiting http://calypso.localhost:3000/ the SVG element:
>
> * has the classes: `gridicon`, `gridicons-reader`, `test-class`, `needs-offset-x`
> * its height and width are 18
> * contains the `myprop` property
> * on click shows an alert

Works like a charm! 💯 

One [final request](https://github.com/Automattic/gridicons/pull/283#discussion_r174398012) (on the `gridicons` repo PR) on my part.

Other than that, LGTM! Thanks a lot for your initiative and persistence! Awesome job!

**Comment by nosolosw** [comment]
https://github.com/Automattic/gridicons/pull/283 has been merged and 3.0.0 published to npm. This has been changed accordingly for the release.

I'm going to merge this in a few hours, after lunch, so I can be around when this lands - just in case I've overlooked something.

**Pull request merged**
